pyOCD/coresight/cortex_m.py

```diff
@@ -335,66 +284,47 @@ def __init__(self, name, bitsize, reg_type, reg_group):
         RegisterInfo('s31',     64,         'float',        'float'),
         ]
 
-    def __init__(self, link, memoryMap=None):
+    def __init__(self, link, dp, ap, memoryMap=None, core_num=0):
         super(CortexM, self).__init__(link, memoryMap)
 
-        self.idcode = 0
-        self.hw_breakpoints = []
-        self.breakpoints = {}
-        self.nb_code = 0
-        self.nb_lit = 0
-        self.num_hw_breakpoint_used = 0
-        self.nb_lit = 0
-        self.fpb_enabled = False
-        self.watchpoints = []
-        self.watchpoint_used = 0
-        self.dwt_configured = False
         self.arch = 0
         self.core_type = 0
         self.has_fpu = False
-        self.part_number = self.__class__.__name__
-        self.dap = Dap(link)
+        self.dp = dp
+        self.ap = ap
+        self.core_number = core_num
+
+        # Set up breakpoints manager.
+        self.fpb = FPB(self.ap)
+        self.dwt = DWT(self.ap)
+        self.sw_bp = SoftwareBreakpointProvider(self)
+        self.bp_manager = BreakpointManager(self)
+        self.bp_manager.add_provider(self.fpb, Target.BREAKPOINT_HW)
+        self.bp_manager.add_provider(self.sw_bp, Target.BREAKPOINT_SW)
 
     def init(self, initial_setup=True, bus_accessible=True):
```

Comment:
The parameters 'initial_setup' and 'bus_accessible' can probably be removed since the init sequence is a lot simpler.